File: Fortran/caesar_cipher.f90
program Caesar_Cipher
  implicit none
 
  ! Initialization
  character(43) :: cipherStr = "Dude Wheres My Car"
  integer :: shift = 4  
 
  write(*, "(2a)") "Welcome to Caesar Ciphers with Fortran!"
  
  ! Call Caesar Cipher Encryption Method
  call encrypt(cipherStr, shift)
  write(*, "(2a)") "Encrypted Cipher Value: ", cipherStr
  
  ! Call Caesar Cipher Decryption Method
  call decrypt(cipherStr, shift)
  write(*, "(2a)") "Decrypted Cipher Value: ", cipherStr
  
  ! Call Caesar Cipher Solve Method
  call solve("DUDE", 26)
 
contains

 
subroutine encrypt(toCipher, shift)
  implicit none
  character(*), intent(inout) :: toCipher
  integer :: shift
  integer :: i
 
  do i = 1, len(toCipher)
    select case(toCipher(i:i))
      case ('A':'Z')
        toCipher(i:i) = achar(modulo(iachar(toCipher(i:i)) - 65 + shift, 26) + 65)
      case ('a':'z')
        toCipher(i:i) = achar(modulo(iachar(toCipher(i:i)) - 97 + shift, 26) + 97)
    end select
  end do
end subroutine
 
 
subroutine decrypt(ciphered, shift)
  implicit none
  character(*), intent(inout) :: ciphered
  integer :: shift
  integer :: reverse
  
  reverse = (shift * (-1))
  call encrypt(ciphered, reverse)
  
end subroutine


subroutine solve(cipher, shiftLoop)
  implicit none
  character(4) :: cipher
  integer :: shiftLoop
  character(4) :: newCiphered
  integer :: i
  
  do i = 0, shiftLoop
    newCiphered = cipher
    call encrypt(newCiphered, i)
    write(*, *) "Caesar ", i, ": ", newCiphered
  end do
  
end subroutine


end program Caesar_Cipher

File: Pascal/caesar_cipher.pas
program CaesarCipher(output);

  var cipherStr: String;
  var shift : Integer;

function encrypt(toCipher: String; shift: Integer): String;
  var
    i: Integer;
  begin
    for i := 1 to length(toCipher) do
      case toCipher[i] of
        'A'..'Z': toCipher[i] := chr(ord('A') + (ord(toCipher[i]) - ord('A') + shift) mod 26);
        'a'..'z': toCipher[i] := chr(ord('a') + (ord(toCipher[i]) - ord('a') + shift) mod 26);
      end;
    encrypt := toCipher;
  end;


function decrypt(ciphered: String; shift: Integer): String;
  begin
    decrypt := encrypt(ciphered, (shift * -1));
  end;


procedure solve(cipher: String; shift: Integer);
  var
    i: Integer;
  begin
    for i := 0 to shift do
      begin
        Writeln('Caesar ', i, ': ', encrypt(cipher, i));
      end;
  end;


// basically main
begin
  // Initialization
  cipherStr := 'Dude Wheres My Car';
  shift := 4;

  Writeln('Welcome to Caesar Ciphers with Pascal!');
  
  // Call Caesar Cipher Encryption Method
  cipherStr := encrypt(cipherStr, shift);
  Writeln('Encrypted Cipher Value: ' + cipherStr);
  
  // Call Caesar Cipher Decryption Method
  cipherStr := decrypt(cipherStr, shift);
  Writeln('Decrypted Cipher Value: ' + cipherStr);
  
  // Call Caesar Cipher Solve Method
  solve('DUDE', 26);
end.